gadget/internal/mkfs.go

```diff
@@ -38,33 +38,43 @@ var (
 	}
 )
 
-// Mkfs creates a filesystem of given type and provided label in the device or file.
-func Mkfs(typ, img, label string) error {
-	return MkfsWithContent(typ, img, label, "")
+// Mkfs creates a filesystem of given type and provided label in the device or
+// file. The device size provides hints for additional tuning of the created
+// filesystem.
+func Mkfs(typ, img, label string, deviceSize uint64) error {
+	return MkfsWithContent(typ, img, label, "", deviceSize)
 }
 
-// Mkfs creates a filesystem of given type and provided label in the device or file.
-// The filesystem is populated with contents of contentRootDir.
-func MkfsWithContent(typ, img, label, contentRootDir string) error {
+// Mkfs creates a filesystem of given type and provided label in the device or
+// file. The filesystem is populated with contents of contentRootDir. The device
+// size provides hints for additional tuning of the created filesystem.
+func MkfsWithContent(typ, img, label, contentRootDir string, deviceSize uint64) error {
 	h, ok := mkfsHandlers[typ]
 	if !ok {
 		return fmt.Errorf("cannot create unsupported filesystem %q", typ)
 	}
-	return h(img, label, contentRootDir)
+	return h(img, label, contentRootDir, deviceSize)
 }
 
 // mkfsExt4 creates an EXT4 filesystem in given image file, with an optional
 // filesystem label, and populates it with the contents of provided root
 // directory.
-func mkfsExt4(img, label, contentsRootDir string) error {
+func mkfsExt4(img, label, contentsRootDir string, deviceSize uint64) error {
 	// Originally taken from ubuntu-image
 	// Switched to use mkfs defaults for https://bugs.launchpad.net/snappy/+bug/1878374
 	// For caveats/requirements in case we need support for older systems:
 	// https://github.com/snapcore/snapd/pull/6997#discussion_r293967140
-	mkfsArgs := []string{
-		"mkfs.ext4",
-		// default usage type
-		"-T", "default",
+	mkfsArgs := []string{"mkfs.ext4"}
+	const size32MiB = 32 * 1024 * 1024
```

Comment:
can't we use gadget.Size here? i.e. this could be

```suggestion
	const size32MiB = 32 * SizeMiB
```

Reply:
Hmm, this has come up a couple of times now. In theory it's ok to do now, but given that it's a `gadget/internal` package, we may run into problems down the road. I can try and look into adding a top level `quantity` package if the name sounds ok, and move the size consts over there. How does that sound?

Reply:
the quantity package sounds great to me, but not high priority or anything